packages/e2e/test/wallet/SingleAddressWallet/txChainHistory.test.ts

```diff
@@ -0,0 +1,76 @@
+import { buildTx } from '@cardano-sdk/wallet';
+import { deepEquals, isNotNil } from '@cardano-sdk/util';
+import { env } from '../environment';
+import { filter, firstValueFrom, map, take } from 'rxjs';
+import { getWallet } from '../../../src';
+import { logger } from '@cardano-sdk/util-dev';
+
+export type TestWallet = Awaited<ReturnType<typeof getWallet>>;
+
+describe('tx chain history', () => {
+  let wallet: TestWallet;
+
+  beforeEach(async () => {
+    wallet = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+  });
+
+  afterEach(() => {
+    wallet.wallet.shutdown();
+  });
+
+  it('submit a transaction and find it in chain history', async () => {
+    const tAdaToSend = 10_000_000n;
+    const saWallet = wallet.wallet;
+
+    await firstValueFrom(saWallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(saWallet.addresses$);
+    const [{ address: receivingAddress }] = await firstValueFrom(saWallet.addresses$);
+
+    logger.info(
+      `Address ${sendingAddress.toString()} will send ${tAdaToSend} lovelace to address ${receivingAddress.toString()}.`
+    );
+
+    // Send 10 tADA to the same wallet.
+    const txBuilder = buildTx({ logger, observableWallet: saWallet });
+    const txOutput = txBuilder.buildOutput().address(receivingAddress).coin(tAdaToSend).toTxOut();
+    const unsignedTx = await txBuilder.addOutput(txOutput).build();
+
+    assertTxIsValid(unsignedTx);
+
+    const signedTx = await unsignedTx.sign();
+    await signedTx.submit();
+
+    logger.info(
+      `Submitted transaction id: ${signedTx.tx.id}, inputs: ${signedTx.tx.body.inputs} and outputs:${signedTx.tx.body.outputs}.`
+    );
+
+    // Wait until wallet two is aware of the funds.
+    await firstValueFrom(saWallet.balance.utxo.total$.pipe(filter(({ coins }) => coins >= tAdaToSend)));
+
+    // Search chain history to see if the transaction is there.
+    const txFoundInHistory = await firstValueFrom(
+      saWallet.transactions.history$.pipe(
+        map((txs) => txs.find((tx) => tx.id === signedTx.tx.id)),
+        filter(isNotNil),
+        take(1)
+      )
+    );
+
+    logger.info(
+      `Found transaction id in chain history: ${txFoundInHistory.id}, inputs: ${txFoundInHistory.body.inputs} and outputs:${txFoundInHistory.body.outputs}.`
+    );
+
+    // Assert
+    expect(txFoundInHistory).toBeDefined();
+    expect(txFoundInHistory.id).toEqual(signedTx.tx.id);
+    expect(txFoundInHistory.body.inputs.length).toEqual(signedTx.tx.body.inputs.length);
+    expect(txFoundInHistory.body.outputs.length).toEqual(signedTx.tx.body.outputs.length);
+    expect(txFoundInHistory.body.inputs.every((txIn, i) => deepEquals(signedTx.tx.body.inputs[i], txIn))).toBe(true);
+    expect(
+      txFoundInHistory.body.outputs.every((txOut, i) => signedTx.tx.body.outputs[i].value.coins === txOut.value.coins)
+    ).toBe(true);
```

Comment:
This would cover other body properties too
```suggestion
    expect(txFoundInHistory.body).toEqual(signedTx.tx.body);
```

Reply:
All the e2e test feedback is applied under 1bbf6d29661f08f7030a02c3316ae7d54b3d4ff3 @mkazlauskas